### PR TITLE
perf improvements over SNAP-2125

### PR DESCRIPTION
**Pull request by kneeraj**
## Changes proposed in this pull request

Corrected some perf mistakes pointed in review comment of SNAP-2125
https://github.com/SnappyDataInc/snappydata/pull/923

## Patch testing

Same test passed.

## ReleaseNotes.txt changes

Need to add newly added properties in documentation.

## Other PRs 

https://github.com/SnappyDataInc/snappydata/pull/923

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/SnappySession.scala [hunk 1]
```diff
@@ -220,7 +220,7 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
   private[sql] var currentKey: SnappySession.CachedKey = _
 
   @transient
-  private[sql] def planCaching: Boolean = sessionState.conf.getConfString(
+  private[sql] var planCaching: Boolean = sessionState.conf.getConfString(
     Property.PlanCaching.name, "true").toBoolean
```
Comment (sumwale) [inline core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: cleaner type-safe way: Property.PlanCaching.get(sessionState.conf)

File: core/src/main/scala/org/apache/spark/sql/SnappySession.scala [hunk 2]
```diff
@@ -1835,7 +1835,7 @@ object SnappySession extends Logging {
   private[this] val ID = new AtomicInteger(0)
   private[sql] val ExecutionKey = "EXECUTION"
 
-  def tokenizationDisabled: Boolean = java.lang.Boolean.getBoolean("DISABLE_TOKENIZATION")
+  var tokenize: Boolean = true
```
Comment (sumwale) [inline core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: should be initialized like for planCaching: Property.Tokenize.get(sessionState.conf)

File: core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala [hunk 3]
```diff
@@ -503,21 +503,31 @@ class SnappyConf(@transient val session: SnappySession)
     case Property.PlanCaching.name =>
       value match {
         case Some(boolval) => {
-          if (boolval.toString.toBoolean) session.clearPlanCache()
+          if (boolval.toString.toBoolean) {
+            session.clearPlanCache()
+          }
+          session.planCaching = boolval.toString.toBoolean
```
Comment (sumwale) [inline core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala]: can store boolVal.toString.toBoolean

File: core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala [hunk 4]
```diff
@@ -503,21 +503,31 @@ class SnappyConf(@transient val session: SnappySession)
...
       }
 
     case Property.PlanCachingAll.name =>
       value match {
         case Some(boolval) => {
-          val disable_tokenization = !boolval.toString.toBoolean
-          System.setProperty("DISABLE_TOKENIZATION", disable_tokenization.toString)
-          if (disable_tokenization) SnappySession.getPlanCache.asMap().clear()
+          val clear_cache = !boolval.toString.toBoolean
+          if (clear_cache) SnappySession.getPlanCache.asMap().clear()
```
Comment (sumwale) [inline core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala]: current session value should be set: session.planCaching = !clear_cache
also naming convention is camel case "clearCache"

File: core/src/main/scala/org/apache/spark/sql/SnappySession.scala (hunk 2 above)
Comment (sumwale) [reply core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: cal also be made private[sql] like for planCaching

Comment (kneeraj) [reply core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: >> should be initialized like for planCaching: Property.Tokenize.get(sessionState.conf)

This is more like a global variable. No session is attached to it at the time of initialization. So kept it as "true". May be this property should be in global sqlconf.
Is there one like that?

File: core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala (hunk 3 above)
Comment (kneeraj) [reply core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala]: Thought but ignored. Hardly matters for this use case.

File: core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala (hunk 4 above)
Comment (kneeraj) [reply core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala]: Does not matter as I am using this session to disable across all sessions. If I do selectively in this session ( like planCaching above ) then this session in some sense is getting differently treated than others.

File: core/src/main/scala/org/apache/spark/sql/SnappySession.scala (hunk 2 above)
Comment (sumwale) [reply core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: Oh, ok. Then Property.Tokenize is better be a SparkConf Val and not SQLVal. Then in SnappyContext.initGlobalSparkContext you could set it: SnappySession.tokenize = Property.Tokenize.get(sc.conf)

**Pull request merged**